extensions/wikia/CuratedContent/CuratedContentSpecialController.class.php

```diff
@@ -187,65 +160,38 @@ public function save() {
 		}
 		$this->response->setFormat( 'json' );
 
-		$sections = $this->request->getArray( 'sections' );
-		list( $sections, $err ) = $this->processSaveLogic( $sections );
+		$helper = new CuratedContentHelper();
 
-		if ( !empty( $err ) ) {
-			$this->response->setVal( 'error', $err );
-			return true;
-		}
+		$sections = $helper->processLogic( $this->request->getArray( 'sections' ) );
+		$errors = ( new CuratedContentValidator( $sections ) )->getErrors();
 
-		$status = WikiFactory::setVarByName( 'wgWikiaCuratedContent', $this->wg->CityId, $sections );
-		$this->response->setVal( 'status', $status );
-
-		if ( $status ) {
-			wfRunHooks( 'CuratedContentSave', [ $sections ] );
-		}
-
-		return true;
-	}
-
-	public function getImage( $id = 0 ) {
-		$file = $this->request->getVal( self::STR_FILE );
-
-		$url = '';
-
-		if ( !empty( $file ) ) {
-			$img = Title::newFromText( $file );
-
-			if ( !empty( $img ) && $img instanceof Title ) {
-				$id = $img->getArticleID();
-			}
-		}
 
-		if ( $id != 0 ) {
-			$is = new ImageServing( [ $id ], 50, 50 );
-			$thumbnail = $is->getImages( 1 );
+		if ( !empty( $errors ) ) {
+			$this->response->setVal( 'error', $errors );
+		} else {
+			$status = WikiFactory::setVarByName( 'wgWikiaCuratedContent', $this->wg->CityId, $sections );
```

Comment:
sometimes you use convention global $wgVar, here you use $this->wg->CityId, could you unify?